var/spack/repos/builtin.mock/packages/both-link-and-build-dep-a/package.py

```diff
@@ -18,7 +18,7 @@ class BothLinkAndBuildDepA(Package):
     homepage = "http://www.example.com"
     url = "http://www.example.com/1.0.tar.gz"
 
-    version("1.0", "0123456789abcdef0123456789abcdef")
+    version("1.0", md5="0123456789abcdef0123456789abcdef")
```

Comment:
do all of these need to be converted?  Do we care?  It'd be nice to have a better way to generate dummy sha256's for tests and just refer to them by name, but I don't really know of a good one that's more clear than writing it out.

Reply:
It's not necessary, but my initial PR was a breaking change that made the signature `version(ver, *, ...)`, since I found a 5 year old comment saying that `checksum=...` as second arg was deprecated, and I realized only half way through that we had md5 checksums too -- i thought it was sha256 everywhere.

In short, I already changed this, then realized the better, non-breaking signature is `version(ver, checksum, *, ...)`.